packages/components/src/accordion/use-accordion.ts

```diff
@@ -52,100 +53,93 @@ export interface UseAccordionProps {
  *
  * @see WAI-ARIA https://www.w3.org/WAI/ARIA/apg/patterns/accordion/
  */
-export function useAccordion(props: UseAccordionProps) {
+export function useAccordion<Multiple extends boolean>(
+  props: UseAccordionProps<Multiple>,
+) {
   const {
     onChange,
-    defaultIndex,
-    index: indexProp,
+    defaultValue,
+    value: valueProp,
     allowMultiple,
     allowToggle,
+    id: idProp,
     ...htmlProps
   } = props
 
   // validate the props and `warn` if used incorrectly
   allowMultipleWarning(props)
   allowMultipleAndAllowToggleWarning(props)
 
-  /**
-   * Think of this as the register to each accordion item.
-   * We used to manage focus between accordion item buttons.
-   *
-   * Every accordion item, registers their button refs in this context
-   */
-  const descendants = useAccordionDescendants()
+  const reactId = useId()
+  const id = `accordion-${idProp ?? reactId}`
 
   /**
    * This state is used to track the index focused accordion
    * button when click on the button, tab on the button, or
    * use the down/up arrow to navigate.
    */
-  const [focusedIndex, setFocusedIndex] = useState(-1)
+  const [focusedId, setFocusedId] = useState<string | null>(null)
```

Comment:
Do we want to call this `focusedValue`?

Reply:
Internally the value will actually map to a unique id:

```
  /**
   * Generate unique ids for all accordion item components (button and panel)
   */
  const reactId = useId()
  const uid = value || reactId

  const buttonId = makeId("button", id, uid)
  const panelId = makeId("panel", id, uid)
```